quark/core/shurikenapkinfo.py

```diff
@@ -0,0 +1,410 @@
+        second_method: MethodObject,
+    ) -> Dict[str, Union[BytecodeObject, str]]:
+        bytecodes = self.get_method_bytecode(parent_method)
+
+        first = self._find_first_bytecode_by_calling_method(
+            bytecodes, first_method
+        )
+        second = self._find_first_bytecode_by_calling_method(
+            bytecodes, second_method
+        )
+
+        return {
+            "first": [
+                first.mnemonic,
+                " ".join(first.registers),
+                first.parameter,
+            ],
+            "first_hex": "",  # TODO - Finish me
+            "second": [
+                second.mnemonic,
+                " ".join(second.registers),
+                second.parameter,
+            ],
+            "second_hex": "",  # TODO - Finish me
+        }
+
+    @property
+    def superclass_relationships(self) -> Dict[str, Set[str]]:
+        hierarchyDict = defaultdict(set)
+
+        for i in range(self.analysis.get_number_of_classes()):
+            rawClass = self.analysis.get_class_by_id(i)
+            className = self._convertClassNameFormat(
+                rawClass.class_name.decode()
+            )
+            superclassName = self._convertClassNameFormat(
+                rawClass.super_class.decode()
+            )
+
+            hierarchyDict[className].add(superclassName)
+
+        return hierarchyDict
+
+    @property
+    def subclass_relationships(self) -> Dict[str, Set[str]]:
+        hierarchyDict = defaultdict(set)
+
+        for i in range(self.analysis.get_number_of_classes()):
+            rawClass = self.analysis.get_class_by_id(i)
+            className = self._convertClassNameFormat(
+                rawClass.class_name.decode()
+            )
+            superclassName = self._convertClassNameFormat(
+                rawClass.super_class.decode()
+            )
+
+            hierarchyDict[superclassName].add(className)
+
+        return hierarchyDict
+
+    def _convert_to_method_object(
+        self,
+        methodAnalysis: hdvmmethodanalysis_t,
+    ) -> MethodObject:
+        className = self._convertClassNameFormat(
+            methodAnalysis.class_name.decode()
+        )
+
+        return MethodObject(
+            # access_flags=methodAnalysis.access_flags,
+            class_name=className,
+            name=methodAnalysis.name.decode(),
+            descriptor=methodAnalysis.descriptor.decode(),
+            cache=methodAnalysis,
+        )
+
+    def _convertClassNameFormat(self, className: str) -> str:
+
+        typeChar = ["V", "Z", "B", "S", "C", "I", "J", "F", "D"]
+
+        patternOne = r"( ?)((?![{}]$)[A-Za-z\./]+)($)".format(
+            "".join(typeChar)
+        )
+        patternTwo = r"(^)([a-zA-Z\./]+)(->)"
+        className = re.sub(patternOne, r"\1L\2;\3", className)
+        className = re.sub(patternTwo, r"\1L\2;\3", className)
+        className = className.replace(".", "/")
+
+        return className
```

Comment:
The function converts the following into the proper formats:
1. Class names, e.g., `a.b.c->` -> `La/b/c;->`
2. Member fields, e.g., `a.b.c->()field1 d.e.f` -> `La/b/c;->()field Ld/e/f;`

Please split this function into two separate functions, so each function handles a single task.